Add an integration test for ambassador addon, fix filename and bump operator version

concaf opened the pull request:
This PR adds an integration test for the ambassador addon as discussed in #8161. Also bumps the operator version.

CC: @priyawadhwa @medyagh

medyagh commented on deploy/addons/ambassador/ambassadorinstallation.yaml:
```diff
@@ -7,3 +7,5 @@ spec:
   helmValues:
     deploymentTool: amb-oper-minikube
+    deploymentStrategy:
+      type: Recreate
```
is this newline needed ?

concaf replied:
Added a newline here :+1:

medyagh commented on pkg/minikube/assets/addons.go:
```diff
@@ -407,7 +407,7 @@ var Addons = map[string]*Addon{
 			"deploy/addons/ambassador/ambassadorinstallation.yaml",
 			vmpath.GuestAddonsDir,
-			"ambassadorinstallation.yaml.yaml",
+			"ambassadorinstallation.yaml",
```
was this a bug that this is trying to fix ? if yes please change the PR title

concaf replied:
updated

medyagh commented:
/ok-to-test

priyawadhwa commented:
Hey @concaf thank you for opening this PR. Are you still working on this? If so could you please:

1. Address review comments
1. Fix the integration tests you wrote; [TestAddons](https://storage.googleapis.com/minikube-builds/logs/8372/3d6b3b3/VirtualBox_Linux.html#fail_TestAddons) is failing across multiple testing environments

concaf commented:
Hey @priyawadhwa, I'm winding up a few things right now, I'll be fixing this next week. Sorry.

priyawadhwa commented:
No worries, totally understand. Just checking in :)

concaf commented:
I've sent https://github.com/kubernetes/minikube/pull/8792 which fixes intermittent failures while deploying ambassador (and dashboard) addon with the `none` driver. Keeping an eye out for the rest of the tests in CI right now.

concaf commented:
Hey @priyawadhwa @medyagh, I've addressed the review comments and fixed the tests. I don't see any failures on "none_Linux" and "KVM_Linux". Failures on "Docker_Linux" and "VirtualBox_Linux" look unrelated. Waiting on other CI results but I think this is ready for another shot ;)

concaf commented:
Gentle nudge :wink:

tstromberg commented:
Fixed merge conflict.

/ok-to-test

medyagh commented:
@concaf sorry for the long delay in the PR review, do you mind pulling master one more time ? feel free to ping me on slack if got delayed again

medyagh commented:
@concaf would you mind please rebasing the PR, otherwise it looks good

medyagh commented:
@concaf do u mind please to solve merge conflicts and try one more time?

tstromberg commented:
Please re-open once merge conflicts are addressed. Thanks!

concaf commented:
Apologies for the delay - I've lost my dev setup to rebase and test this and I don't have the bandwidth to fix this right now.
CC: @kflynn @khussey do you folks have someone to do a quick rebase and test this - should not take long.
